cmd/vela-worker/exec.go

```diff
@@ -93,63 +93,58 @@ func (w *Worker) exec(index int) error {
 		t = time.Duration(item.Repo.GetTimeout()) * time.Minute
 	}
 
-	// create a background context
-	ctx := context.Background()
+	// create a build context
+	buildCtx, buildDone := context.WithCancel(ctx)
+	defer buildDone()
```

Comment:
this change means that a worker shutdown is going to cancel the context instead of just killing things.

If we don't want to do this, then we should expand the comment to explain why not.